code/modules/vending/npc.dm

```diff
@@ -0,0 +1,150 @@
+	icon = 'icons/mob/simple_human.dmi'
+	icon_state = "faceless"
+	interaction_flags_atom = INTERACT_ATOM_ATTACK_HAND
+	layer = MOB_LAYER
+	///Corpse spawned when vendor is deconstructed (MURDERED)
+	var/corpse = /obj/effect/mob_spawn/human/corpse
+	///Phrases used when you talk to the NPC
+	var/list/lore = list("Hello! I am the test NPC.",
+						"Man, shut the fuck up."
+						)
+	///List of items able to be sold to the npc
+	var/list/wanted_items = list()
+
+/obj/machinery/vending/npc/Initialize()
+	. = ..()
+	QDEL_NULL(wires)
+	QDEL_NULL(coin)
+	QDEL_NULL(bill)
+	QDEL_NULL(Radio)
+
+/obj/machinery/vending/npc/attackby(obj/item/I, mob/user, params)
+	return
+
+/obj/machinery/vending/npc/crowbar_act(mob/living/user, obj/item/I)
+	return
+
+/obj/machinery/vending/npc/wrench_act(mob/living/user, obj/item/I)
+	return
+
+/obj/machinery/vending/npc/screwdriver_act(mob/living/user, obj/item/I)
+	return
+
+/obj/machinery/vending/npc/deconstruct(disassembled = TRUE)
+	if(corpse)
+		new corpse(src)
+	qdel(src)
+
+/obj/machinery/vending/npc/loadingAttempt(obj/item/I, mob/user)
+	return
+
+/obj/machinery/vending/npc/emag_act(mob/user)
+	return
+
+/obj/machinery/vending/npc/hitby(atom/movable/AM, skipcatch, hitpush, blocked, datum/thrownthing/throwingdatum)
+	if(istype(AM, /obj/item))
+		return
+	..()
+
+/obj/machinery/vending/npc/interact(mob/user)
+	face_atom(user)
+	var/list/npc_options = list(
+		"Buy" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_buy"),
+		"Sell" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_sell"),
+		"Talk" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_talk")
+		)
+	var/npc_result = show_radial_menu(user, src, npc_options, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(!check_menu(user))
+		return
+	switch(npc_result)
+		if("Buy")
+			return ui_interact(user)
+		if("Sell")
+			return try_sell(user)
+		if("Talk")
+			return deep_lore(user)
+	face_atom(user)
+	return FALSE
+
+/obj/machinery/vending/npc/ui_act(action, params)
+	. = ..()
+	face_atom(usr)
+
+/obj/machinery/vending/npc/proc/check_menu(mob/user)
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated() || !user.Adjacent(src))
+		return FALSE
+	return TRUE
+
+/obj/machinery/vending/npc/proc/try_sell(mob/user)
+	var/obj/item/activehanditem = user.get_active_held_item()
+	var/obj/item/inactivehanditem = user.get_inactive_held_item()
+	if(is_type_in_list(activehanditem, wanted_items))
+		if(istype(activehanditem, /obj/item/stack))
+			var/obj/item/stack/stackoverflow = activehanditem
+			generate_cash((wanted_items[stackoverflow.type]) * stackoverflow.amount, user)
+			stackoverflow.use(stackoverflow.amount)
+			return
+		generate_cash(wanted_items[activehanditem.type], user)
+		return
+	else if (is_type_in_list(inactivehanditem, wanted_items))
+		generate_cash(wanted_items[inactivehanditem.type], user)
+		qdel(inactivehanditem)
+		return
```

Comment:
what if the offhand is a stack

i would redesign this so that you have a proc ``proc_that_actually_sells_item(item)``, and have it return TRUE or FALSE on if it actually sold the item. check with active hand, if it returns FALSE then check with the item in the inactive hand

reduces code duplication and should improve legibility/maintainability

```
try_sell(user):
  if (!sell_item(active_item))
    sell_item(inactive_item)
```

Reply:
actually with the say statement below, you could even do something like

```
if (!(sell_item(active_item) || sell_item(inactive_item)))
  say(blah blah)
```

though selling the item in the offhand may be unexpected to the player, maybe have a confirmation that they want to sell it that also tells them how much they'll get paid for it?